app/views/admin/applications/show.html.erb

```diff
@@ -0,0 +1,15 @@
+<h1>Admin Applications</h1> 
+<% @application.pets.each do |pet| %>
+  <%= form_with url: "/pet_applications/#{@application.get_pet_app(pet.id).id}?app_id=#{@application.id}&pet_id=#{pet.id}", method: :patch, local: true do |f|  %>
```

Comment:
This is a nice RESTful route as it names both resources involved in the join. However it can be a pain to have to add all those query params so another equally RESTful option is: "/applications/:app_id/pets/:pet_id"